controllers/v1/criteria.js

```diff
@@ -0,0 +1,1319 @@
+const config = require('../../config/config');
+const rp = require('request-promise');
+const request = require('request');
+const model = require('../../db');
+const helperFunc = require('../../helper/chart_data');
+const pdfHandler = require('../../helper/common_handler');
+const omit = require('object.omit');
+const url = require("url");
+const rimraf = require("rimraf");
+const fs = require('fs');
+const path = require('path');
+
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceReport 
+   * Instance report
+   * @apiVersion 1.0.0
+   * @apiGroup Criteria
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "submissionId": "",
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*     {
+       "entityName": "",
+       "observationName": "",
+       "observationId": "",
+       "entityType": "",
+       "entityId": "",
+       "response": [{
+         "criteriaName": "",
+         "criteriaId": "",
+         "questionArray": [{
+            "order": "",
+            "question": "",
+            "responseType": "",
+            "answers": [],
+            "chart": {},
+            "instanceQuestions":[],
+            "evidences":[
+              {"url":"", "extension":""},
+            ]
+         }]
+         
+       }]
+*     }
+   * @apiUse errorBody
+   */
+
+//Controller for instance observation report
+exports.instanceReport = async function (req, res) {
+
+    let data = await instanceReportData(req, res);
+
+    res.send(data);
+}
+
+
+async function instanceReportData(req, res) {
+
+    return new Promise(async function (resolve, reject) {
+
+        if (!req.body.submissionId) {
+            let response = {
+                result: false,
+                message: 'submissionId is a required field'
+            };
+            resolve(response);
+        } else {
+
+            let submissionId = req.body.submissionId;
+
+            model.MyModel.findOneAsync({ qid: "instance_criteria_report_query" }, { allow_filtering: true })
+                .then(async function (result) {
+
+                    let bodyParam = JSON.parse(result.query);
+
+                    if (config.druid.observation_datasource_name) {
+                        bodyParam.dataSource = config.druid.observation_datasource_name;
+                    }
+
+                    //if filter is given
+                    if (req.body.filter) {
```

Comment:
@deepakr123 - Make it as below : -

{
  "submissionId": "",
  "filter": {
      "criteria": [// array of criteria id]
}
}

Reply:
done